File: Daily Challanges/2026-03-March/2026-03-21.js
function rotateGrid(grid) {
  return Array.from({ length: 6 }, (_, i) =>
    Array.from({ length: 6 }, (_, j) => grid[5 - j][i]).join(""),
  );
}

function isValid(grid) {
  function isBlock(i, j) {
    for (let x = 0; x < 2; x++) {
      for (let y = 0; y < 2; y++) {
        if (grid[i + x][j + y] !== "1") return false;
      }
    }
    return true;
  }
  return isBlock(0, 0) && isBlock(0, 4) && isBlock(4, 0);
}

function decodeQr(qrCode) {
  for (let k = 0; k < 4; k++) {
    if (isValid(qrCode)) {
      let result = [];
      for (let i = 0; i < 6; i++) {
        for (let j = 0; j < 6; j++) {
          if ((i < 2 && j < 2) || (i < 2 && j >= 4) || (i >= 4 && j < 2)) {
            continue;
          }
          result.push(qrCode[i][j]);
        }
      }
      return result.join("");
    }
    qrCode = rotateGrid(qrCode);
  }
}

console.log(
  decodeQr(["110011", "110011", "000000", "000000", "110000", "110001"]),
);

File: Daily Challanges/2026-03-March/2026-03-21.py
def rotate(g):
    return ["".join(g[5-j][i] for j in range(6)) for i in range(6)]

def valid(g):
    def block(i, j):
        return all(g[i+x][j+y] == '1' for x in range(2) for y in range(2))
    return block(0,0) and block(0,4) and block(4,0)


def decode_qr(qr_code):
    for _ in range(4):
        if valid(qr_code):
            result = []
            for i in range(6):
                for j in range(6):
                    if (i < 2 and j < 2) or (i < 2 and j >= 4) or (i >= 4 and j < 2):
                        continue
                    result.append(qr_code[i][j])
            return "".join(result)
        qr_code = rotate(qr_code)

print(decode_qr(["110011", "110011", "000000", "000000", "110000", "110001"]))